src/graph/PropertyRetriever.test.ts

```diff
@@ -29,4 +30,12 @@ describe('PropertyRetriever', () => {
     expect(uut().retrieve('singletonNumber')).toBe(1);
     expect(uut().retrieve('singletonNumber')).toBe(1);
   });
+
+  it('should throw error if the dependency is undefined', () => {
+    const graph = new MissingDependencyGraph();
+    const uut1 = new PropertyRetriever(graph);
+    const errorMessage = 'Could not resolve dependency aString '
```

Comment:
1. Let's add another test where subgraphs are defined
2. Let's make sure that when there are no subgraphs, the error message prints: `...or in any of its subgraphs: []`